riscv/csrs.cc

```diff
@@ -798,8 +817,19 @@ masked_csr_t::masked_csr_t(processor_t* const proc, const reg_t addr, const reg_
   mask(mask) {
 }
 
+void masked_csr_t::write_full(const reg_t val) noexcept {
+  bool success = basic_csr_t::unlogged_write((read() & ~mask) | (val & mask));
+  if (success)
+    log_write();
+}
```

Comment:
This is a bit messy, having to deal with logging specially here. This also breaks logging of `csrw mstatush` which now logs `mstatus` (the low 32 bits) instead of `mstatush`.

See #1045 for my alternative proposal. (Your review appreciated!)

Reply:
I have updated based on the new rv32_low/high_csr_t way.